test/integration/main_test.go

```diff
@@ -0,0 +1,416 @@
+			filePath: fmt.Sprintf("../../%s/tool.yaml", cfg.WorkingDir),
+			wantOutput: strings.ReplaceAll(`
+------Successfully Completed Commands------
+- gcloud projects list --format json --uri --filter projectId:INTEG_TEST_PROJECT_ID
+- gcloud projects list --uri --filter projectId:INTEG_TEST_PROJECT_ID`,
+				"INTEG_TEST_PROJECT_ID", cfg.ProjectID),
+		},
+		{
+			name:     "success_verbose",
+			filePath: fmt.Sprintf("../../%s/tool.yaml", cfg.WorkingDir),
+			verbose:  true,
+			wantOutput: strings.ReplaceAll(`------Tool Commands Output------
+gcloud projects list --format json --uri --filter projectId:INTEG_TEST_PROJECT_ID
+[
+  "https://cloudresourcemanager.googleapis.com/v1/projects/INTEG_TEST_PROJECT_ID"
+]
+
+gcloud projects list --uri --filter projectId:INTEG_TEST_PROJECT_ID
+https://cloudresourcemanager.googleapis.com/v1/projects/INTEG_TEST_PROJECT_ID
+------Successfully Completed Commands------
+- gcloud projects list --format json --uri --filter projectId:INTEG_TEST_PROJECT_ID
+- gcloud projects list --uri --filter projectId:INTEG_TEST_PROJECT_ID
+`, "INTEG_TEST_PROJECT_ID", cfg.ProjectID),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			args := []string{
+				"tool", "cleanup",
+				"-path", tc.filePath,
+			}
+			if tc.verbose {
+				args = append(args, "-verbose")
+			}
+
+			_, stdout, stderr, err := cli.PipeAndRun(ctx, args)
+			if err != nil {
+				t.Fatalf("Process(%+v) failed to run command %v", tc.name, err)
+			}
+
+			if strings.TrimSpace(tc.wantOutput) != strings.TrimSpace(stdout.String()) {
+				t.Errorf("Process(%+v) output response got %q, want %q)", tc.name, stdout.String(), tc.wantOutput)
+			}
+			if stderr.String() != "" {
+				t.Errorf("Process(%+v) got unexpected error: %q)", tc.name, stderr.String())
+			}
+		})
+	}
+}
+
+func TestToolValidate(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name       string
+		filePath   string
+		wantOutput string
+	}{
+		{
+			name:       "success",
+			filePath:   fmt.Sprintf("../../%s/tool.yaml", cfg.WorkingDir),
+			wantOutput: "Successfully validated tool request",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			args := []string{
+				"tool", "validate",
+				"-path", tc.filePath,
+			}
+
+			_, stdout, stderr, err := cli.PipeAndRun(ctx, args)
+			if err != nil {
+				t.Fatalf("Process(%+v) failed to run command %v", tc.name, err)
+			}
+
+			if strings.TrimSpace(tc.wantOutput) != strings.TrimSpace(stdout.String()) {
+				t.Errorf("Process(%+v) output response got %q, want %q)", tc.name, stdout.String(), tc.wantOutput)
+			}
+			if stderr.String() != "" {
+				t.Errorf("Process(%+v) got unexpected error: %q)", tc.name, stderr.String())
+			}
+		})
+	}
+}
+
+func testGetPolicyBindings(ctx context.Context, tb testing.TB, cfg *config) ([]*iampb.Binding, error) {
```

Comment:
Do not return error. Instead, use `tb.Fatal` to fail

Reply:
Might as well merge `testGotBindings` into this func

Reply:
Sure, updating